static/js/buddy_data.js

```diff
@@ -247,30 +247,79 @@ function maybe_shrink_list(user_ids, filter_text) {
 
 exports.get_filtered_and_sorted_user_ids = function (filter_text) {
     var user_ids;
-
-    if (filter_text) {
-        // If there's a filter, select from all users, not just those
-        // recently active.
-        user_ids = filter_user_ids(filter_text, people.get_active_user_ids());
-    } else {
-        // From large realms, the user_ids in presence may exclude
-        // users who have been idle more than three weeks.  When the
-        // filter text is blank, we show only those recently active users.
-        user_ids = presence.get_user_ids();
+    var use_member_list = page_params.user_list_content === "member";
+    var use_pm_with_filter = false;
+    var use_stream_filter = false;
+    var filter;
+    if (use_member_list) {
+        var active = narrow_state.active();
+        filter = narrow_state.filter();
+        if (active) {
+            use_pm_with_filter = filter.has_operator('pm-with');
+            use_stream_filter = filter.has_operator('stream');
+        }
     }
-
-    user_ids = _.filter(user_ids, function (user_id) {
-        var person = people.get_person_from_user_id(user_id);
-
-        if (person) {
-            // if the user is bot, do not show in presence data.
-            if (person.is_bot) {
-                return false;
+    // If or condition evaluates to force, follow normal "all user" path
+    if (use_member_list && (use_pm_with_filter || use_stream_filter)) {
+        if (use_pm_with_filter) {
+            var user_emails = filter.operands('pm-with')[0].split(',');
+            user_ids = _.reduce(user_emails, function (memo, user_email) {
```

Comment:
What is the `memo` variable name supposed to mean?

Reply:
That's just the typical name for the first parameter of the call back for `_.reduce`, it's what the underscore docs use but I understand that it's a bit confusing.

1. I could add a link to the docs in a comment (although that wouldn't be helpful because they don't actually give an example of constructing an array).
2. I could link to https://stackoverflow.com/a/30804674 for where I got the idea from.
3. Or I could rename `memo` to `accumulator`.

what would be the most sensible option?